### ISSUE TITLE
Vivaldi browser: auto-lock fails

### ISSUE THREAD

**Issue by slalik** [palant/pfp#99]
I use PfP in the Vivaldi browser (Linux, 64 bit). Everything works except the auto-lock with positive delay. The delay 0 works. Also manual "Lock passwords" works. Restarting browser locks as it should.

On the same computer for Firefox and Google Chrome auto-lock works.

**Comment by palant** [comment]
Confirmed in 2.5.1525.43 on Linux.

**Comment by palant** [comment]
It's a weird quirk in Vivaldi - while the pop-up disappears, it doesn't actually unload. So the pop-up's port doesn't disconnect, there is no `pagehide` or `unload` event. AFAICT there is no way of telling that the pop-up is no longer visible. And since as far as PfP is concerned the pop-up is still active, autolock won't kick in. Only when you open the pop-up again the previous pop-up instance unloads which allows autolock to work - only if delay is 0 of course, because the new instance loads immediately after that.

I have no idea why it has been implemented in this way, it could simply be a bug. I filed a Vivaldi bug report VB-53559 on this.
